src/RequestParser.php

```diff
@@ -28,12 +28,40 @@ public function __construct(callable $readParameter, $config)
             $config = new Config();
             $config->setExceptionMessageFactory(new LegacyExceptionMessageFactory());
             $config->setExceptionFactory(new LegacyExceptionFactory($exceptionFactory));
+
+            trigger_error($this->generateCallbackBasedExceptionHandlingDeprecatedMessage($exceptionFactory), E_USER_DEPRECATED);
```

Comment:
How this looks inside the server log:
```
[Tue Aug  2 21:36:32 2016] PHP Deprecated:  RequestParser: Callback based Exception configuration is deprecated. To get get better error messages for invalid parameters (e.g. a string is provided to a `->int()->required()`) you just need to migrate your code (more infos: http://bit.ly/2aGyX99).

Your current error callback (`function ($parameterName) { return new CustomException("..."); }`) should be replaced with something along of:
class CustomExceptionMessageFactory extends \MPScholten\RequestParser\ExceptionMessageFactory {
    protected function createNotFoundMessage($parameterName) { return "param $parameterName is missing"; }
}

// In your controller:
$config = new \MPScholten\RequestParser\Config();
$config->setExceptionFactory(new CustomExceptionMessageFactory());
$this->initRequestParser($request, $config);
 in /Users/marc/request-parser/src/RequestParser.php on line 32
```

To try it out yourself just take e.g. the`examples/custom-exception.php` and add a legacy exception handler:
```php
$this->initRequestParser($request, function($e) {
    return new CustomException("param $e is missing");
});
```

and then trigger a page reload...

Okay speaking of page reload, maybe we shouldn't do this logging. If we log this for each request handled by the library, it might blow up someones error log (100 page views => 100x the above log message). Maybe we should only log it when the legacy exception handler is used (so only when a parameter is missing or invalid), or maybe we should not log it at all. Not sure :)